### PR TITLE
style(Table): Add the rowScope style

### LINKED ISSUE #39688: 补充 Table rowScope 用法 demo
ref: https://github.com/react-component/table/pull/859

### LINKED ISSUE #40297: table 组件的 rowScope 属性会造成一些样式问题，目前发现的是 border 消失
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/demo-of-rowscope-690ky2)

### Steps to reproduce

打开链接即可

### What is expected?

rowScope 不造成任何副作用

### What is actually happening?

table 组件的 rowScope 属性会造成一些样式问题，目前发现的是 border 消失

| Environment | Info |
| --- | --- |
| antd | 5.1.5 |
| React | 18 |
| System | Mac |
| Browser | chrome101 |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/40297
close https://github.com/ant-design/ant-design/issues/39688

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Add rowScope style for table         |
| 🇨🇳 Chinese |  增加 table 中 rowScope 的样式        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by Yuiai01** [comment]
为了更加符合行表头，我将右边竖线，以及默认的 text-align: start 去除了，不知道这样行不行，如图：
image

**Comment by MadCcc** [comment]
错位了

**Comment by Yuiai01** [comment]
> 错位了

我再试试，大佬能大概指一下吗

**Comment by Yuiai01** [comment]
现在的样式是酱紫的：
image

**Comment by MadCcc** [comment]
Closing since style has been reverted: https://github.com/ant-design/ant-design/pull/40469

**Comment by afc163** [comment]
@Yuiai01 这个还得加，现在样式还是有问题 https://codesandbox.io/s/demo-of-rowscope-forked-m7x1xy?file=/package.json

**Comment by afc163** [comment]
搞好了，https://github.com/ant-design/ant-design/pull/40271 也能合掉。 cc @MuxinFeng

File: components/table/style/index.ts [hunk 1]
```diff
@@ -195,19 +196,25 @@ const genTableStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
             },
           },
 
+          '> th': {
+            textAlign: 'start',
```
Comment (Yuiai01) [inline components/table/style/index.ts:200]: 这里覆盖掉 th 默认的 `text-align: center` 是由于表头默认左对齐，当列宽很大时，就会出现很诡异的一幕，如下图：
image

**Comment by MadCcc** [comment]
最好把 demo 合进来，CR 有点困难。

**Comment by MuxinFeng** [comment]
我不知道咋往你仓库提 pr，要不你补个 demo 吧，听偏右大佬的意思时不用单独开个 demo，在原有的 demo 上加个属性试试？
可以参考下这个 https://codesandbox.io/s/demo-of-rowscope-690ky2

**Comment by MuxinFeng** [comment]
demo 的 pr 提到你的分支啦，求看看~ @Yuiai01

**Review by MadCcc — APPROVED** [review]

**Pull request merged**